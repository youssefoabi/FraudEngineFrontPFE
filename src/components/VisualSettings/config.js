export const ImagesUrlConfig = [
  {
    columns: 5,
    id: 'urlImageDesktop',
    type: 'input',
    label: 'URL image PC',
  },
  {
    columns: 5,
    id: 'urlLinkDesktop',
    type: 'input',
    label: 'Lien PC',
  },
  {
    columns: 5,
    id: 'urlImageMobile',
    type: 'input',
    label: 'URL image Mobile',
  },
  {
    columns: 5,
    id: 'urlLinkMobile',
    type: 'input',
    label: 'Lien Mobile',
  },
];

export const AlternativeTextConfig = [
  {
    columns: 5,
    id: 'alt',
    type: 'input',
    label: 'ALT',
  },
  {
    columns: 4,
    id: 'openInNewTab',
    type: 'switch',
    label: 'Ouvrir dans un nouvel onglet',
  },
  {
    columns: 3,
    id: 'isEnable',
    type: 'switch',
    label: 'Active',
  },
];

export const MarketingTextConfig = [
  {
    columns: 8,
    id: 'marketingText',
    type: 'input',
    label: "Texte de l'accroche marketing",
    size: 'large',
    multiline: true,
  },
  {
    columns: 3,
    id: 'isMarketingTextEnable',
    type: 'switch',
    label: 'Active',
  },
  {
    columns: 3,
    id: 'searchText',
    type: 'input',
    label: 'Texte du bouton rechercher',
  },
];
