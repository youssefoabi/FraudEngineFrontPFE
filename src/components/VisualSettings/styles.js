import { black, cornflower, ironGrey, white } from '@cdiscount/ui-colors';
import { fontSizeBase } from '@cdiscount/ui-fonts';
import { isNilOrEmpty } from 'ramda-adjunct';

export default function style() {
  return {
    headerSlider: {
      padding: '17px 30px 12px',
      borderBottom: `1px solid ${cornflower}`,
      borderRadius: '5px',
      background: white,
      color: black,
      fontSize: fontSizeBase,
      fontWeight: 300,
      fontFamily: '"Open Sans", sans-serif',
      textTransform: 'uppercase',
    },
    imagesWrapper: {
      flex: 1,
    },
    imagePreview: {
      height: 74,
      marginTop: '10px',
      width: 280,
      flex: 1,
      backgroundRepeat: 'no-repeat',
      backgroundSize: '100% 100%',
      backgroundColor: ironGrey,
    },
    imageUrlDesktop: ({ urlImageDesktop }) =>
      !isNilOrEmpty(urlImageDesktop) ? { backgroundImage: `url(${urlImageDesktop})` } : {},
    imageUrlMobile: ({ urlImageMobile }) =>
      !isNilOrEmpty(urlImageMobile) ? { backgroundImage: `url(${urlImageMobile})` } : {},
  };
}
