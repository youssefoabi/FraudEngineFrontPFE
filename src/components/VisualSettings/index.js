import { DataEdit, ExpansionPanel, GridItem, useStyles } from '@cdiscount/backoffice-ui';
import { FormLabel, Grid } from '@material-ui/core';
import { func } from 'prop-types';
import { pathOr } from 'ramda';
import clsx from 'clsx';
import React from 'react';

import { AlternativeTextConfig, ImagesUrlConfig, MarketingTextConfig } from './config';
import { enginePropTypes } from '../EngineForm/enginePropTypes';
import styles from './styles';

const VisualSettings = ({ engine, handleChange }) => {
  const backgroundImages = pathOr([], ['backgroundImages'], engine);
  const backgroundImage = pathOr({}, [0], backgroundImages);
  const urlImageDesktop = pathOr('', ['urlImageDesktop'], backgroundImage);
  const urlImageMobile = pathOr('', ['urlImageMobile'], backgroundImage);

  const handleBackgroundImageChange = ({ id, value }) => {
    handleChange({
      id: 'backgroundImages',
      value: { ...backgroundImages, 0: { ...backgroundImage, [id]: value } },
    });
  };

  const classes = useStyles(styles, { urlImageDesktop, urlImageMobile });

  return (
    <ExpansionPanel title="Gestion du visuel">
      <Grid container direction="column">
        <div className={classes.headerSlider}>Slider d&apos;images de fond</div>
        <Grid container spacing={2}>
          <GridItem item xs={12} sm={12} md={3} xl={3}>
            <div className={classes.imagesWrapper}>
              <FormLabel>Aperçu PC</FormLabel>
              <div className={clsx(classes.imagePreview, classes.imageUrlDesktop)} alt="" />
              <FormLabel>Aperçu Mobile</FormLabel>
              <div className={clsx(classes.imagePreview, classes.imageUrlMobile)} alt="" />
            </div>
          </GridItem>
          <GridItem item xs={12} sm={12} md={9} xl={9}>
            <Grid>
              <DataEdit
                fields={ImagesUrlConfig}
                values={backgroundImage}
                onChange={handleBackgroundImageChange}
              />
            </Grid>
            <Grid>
              <DataEdit
                fields={AlternativeTextConfig}
                values={backgroundImage}
                onChange={handleBackgroundImageChange}
              />
            </Grid>
          </GridItem>
        </Grid>
        <br />
        <GridItem title="Parametrages des textes">
          <Grid item xs={12} md={12} lg={12}>
            <DataEdit fields={MarketingTextConfig} values={engine} onChange={handleChange} />
          </Grid>
        </GridItem>
      </Grid>
    </ExpansionPanel>
  );
};

VisualSettings.propTypes = {
  handleChange: func,
  engine: enginePropTypes,
};

VisualSettings.defaultProps = {
  handleChange: Function.prototype,
  engine: null,
};

export default VisualSettings;
