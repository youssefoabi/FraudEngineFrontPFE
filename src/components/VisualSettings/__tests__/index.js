import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import VisualSettings from '../';

describe(VisualSettings.name, () => {
  itRendersCorrectly(<VisualSettings />);
});
