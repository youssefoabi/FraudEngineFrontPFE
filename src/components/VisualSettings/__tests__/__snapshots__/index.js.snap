// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualSettings renders correctly 1`] = `
<p
  defaultExpanded={false}
  title="Gestion du visuel"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="column"
  >
    <div
      className="makeStyles-headerSlider-1"
    >
      Slider d'images de fond
    </div>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <h
        fullheight={false}
        item={true}
        md={3}
        sm={12}
        title={null}
        xl={3}
        xs={12}
      >
        <div
          className="makeStyles-imagesWrapper-2"
        >
          <WithStyles(ForwardRef(FormLabel))>
            Aperçu PC
          </WithStyles(ForwardRef(FormLabel))>
          <div
            alt=""
            className="makeStyles-imagePreview-3 makeStyles-imageUrlDesktop-4 makeStyles-imageUrlDesktop-6"
          />
          <WithStyles(ForwardRef(FormLabel))>
            Aperçu Mobile
          </WithStyles(ForwardRef(FormLabel))>
          <div
            alt=""
            className="makeStyles-imagePreview-3 makeStyles-imageUrlMobile-5 makeStyles-imageUrlMobile-7"
          />
        </div>
      </h>
      <h
        fullheight={false}
        item={true}
        md={9}
        sm={12}
        title={null}
        xl={9}
        xs={12}
      >
        <WithStyles(ForwardRef(Grid))>
          <f
            config={null}
            fields={
              Array [
                Object {
                  "columns": 5,
                  "id": "urlImageDesktop",
                  "label": "URL image PC",
                  "type": "input",
                },
                Object {
                  "columns": 5,
                  "id": "urlLinkDesktop",
                  "label": "Lien PC",
                  "type": "input",
                },
                Object {
                  "columns": 5,
                  "id": "urlImageMobile",
                  "label": "URL image Mobile",
                  "type": "input",
                },
                Object {
                  "columns": 5,
                  "id": "urlLinkMobile",
                  "label": "Lien Mobile",
                  "type": "input",
                },
              ]
            }
            inputs={null}
            onChange={[Function]}
            onInit={[Function]}
            title={null}
            values={Object {}}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))>
          <f
            config={null}
            fields={
              Array [
                Object {
                  "columns": 5,
                  "id": "alt",
                  "label": "ALT",
                  "type": "input",
                },
                Object {
                  "columns": 4,
                  "id": "openInNewTab",
                  "label": "Ouvrir dans un nouvel onglet",
                  "type": "switch",
                },
                Object {
                  "columns": 3,
                  "id": "isEnable",
                  "label": "Active",
                  "type": "switch",
                },
              ]
            }
            inputs={null}
            onChange={[Function]}
            onInit={[Function]}
            title={null}
            values={Object {}}
          />
        </WithStyles(ForwardRef(Grid))>
      </h>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <h
      fullheight={false}
      title="Parametrages des textes"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={12}
        md={12}
        xs={12}
      >
        <f
          config={null}
          fields={
            Array [
              Object {
                "columns": 8,
                "id": "marketingText",
                "label": "Texte de l'accroche marketing",
                "multiline": true,
                "size": "large",
                "type": "input",
              },
              Object {
                "columns": 3,
                "id": "isMarketingTextEnable",
                "label": "Active",
                "type": "switch",
              },
              Object {
                "columns": 3,
                "id": "searchText",
                "label": "Texte du bouton rechercher",
                "type": "input",
              },
            ]
          }
          inputs={null}
          onChange={[Function]}
          onInit={[Function]}
          title={null}
          values={null}
        />
      </WithStyles(ForwardRef(Grid))>
    </h>
  </WithStyles(ForwardRef(Grid))>
</p>
`;
